src/lib/AST/ASTVisitor.cpp

```diff
@@ -2101,6 +2187,98 @@ class ASTVisitor
 
     //------------------------------------------------
 
+    void
+    buildNamespaceAlias(
+        AliasInfo& I,
+        bool created,
+        NamespaceAliasDecl* D)
+    {
+        bool documented = parseRawComment(I.javadoc, D);
+        addSourceLocation(I, D->getBeginLoc(), true, documented);
+
+        if(! created)
+            return;
+
+        I.Name = extractName(D);
+        I.FullyQualifiedName = buildNameInfo(D);
+
+        // A NamedDecl nominated by a NamespaceAliasDecl
+        // will be one of the following:
+        // - NamespaceDecl
+        if(NamedDecl* ND = D->getAliasedNamespace())
+        {
+            SymbolID id;
+            getDependencyID(ND, id);
+            I.AliasedSymbol = id;
+        }
+
+        getParentNamespaces(I, D);
+    }
+
+
+    //------------------------------------------------
+
+    void
+    buildUsingDirective(
+        UsingInfo& I,
+        bool created,
+        UsingDirectiveDecl* D)
+    {
+        bool documented = parseRawComment(I.javadoc, D);
+        addSourceLocation(I, D->getBeginLoc(), true, documented);
+
+        if(! created)
+            return;
+
+        I.Name = extractName(D);
+        I.Class = UsingClass::Namespace;
+
+        if (D->getQualifier())
+        {
+            I.Qualifier = buildNameInfo(D->getQualifier());
+        }
+
+        if (NamedDecl* ND = D->getNominatedNamespace())
+        {
+            SymbolID id;
+            getDependencyID(ND, id);
+            I.UsingSymbols.emplace_back(id);
+        }
+        getParentNamespaces(I, D);
+    }
+
+
+    //------------------------------------------------
+
+    void
+    buildUsingDeclaration(
+        UsingInfo& I,
+        bool created,
+        UsingDecl* D)
+    {
+        bool documented = parseRawComment(I.javadoc, D);
+        addSourceLocation(I, D->getBeginLoc(), true, documented);
+
+        if(! created)
+            return;
+
+        I.Name = extractName(D);
+        I.Class = UsingClass::Normal;
+        I.Qualifier = buildNameInfo(D->getQualifier());
+
+        for (auto const* shadow : D->shadows())
```

Comment:
Nit pick, but this could just be  
```cpp
for (auto const* UDS : D->shadows())
    getDependencyID(UDS->getTargetDecl(), I.UsingSymbols.emplace_back());
```